### PR TITLE
Fix memory leak caused by fuzzyfilter

### PR DESCRIPTION
**Pull request by jensenojs**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #14081

## What this PR does / why we need it:

free a vector that missed before

### CODE REVIEW

**Comment by matrix-meow** [comment]
@jensenojs Thanks for your contributions!

Title: free pass2RuntimeFilter

Body: This pull request fixes issue #14081. It adds code to free a vector that was not being freed before.

Changes:
- In the `Free` method of the `Argument` struct in `types.go`, a check is added to free the `pass2RuntimeFilter` vector if it is not `nil`.

Review:

1. Security: There are no security issues in this pull request.

2. Code Quality: The changes made in this pull request seem to be addressing a bug by freeing a vector that was not being freed before. The code changes look fine and should resolve the issue.

Suggestions for optimization:

1. It would be helpful to add a comment explaining why the `pass2RuntimeFilter` vector needs to be freed and any potential side effects of not freeing it.

2. Consider adding a test case to verify that the vector is correctly freed when the `Free` method is called.

Overall, the changes made in this pull request seem to be necessary and should resolve the bug. The code changes are minimal and do not introduce any new issues.

**Comment by jensenojs** [comment]
fuzzy filter造成的内存泄漏是因为之前漏释放了一个vector, 在修改之后make debug已经检验不出来自fuzzy的内存泄漏

image

**Review by m-schen — APPROVED** [review]

File: pkg/sql/colexec/fuzzyfilter/types.go [hunk 1]
```diff
@@ -113,6 +113,9 @@ func (arg *Argument) Free(proc *process.Process, pipelineFailed bool, err error)
 		arg.rbat.Clean(proc.GetMPool())
 		arg.rbat = nil
 	}
+	if arg.pass2RuntimeFilter != nil {
+		proc.PutVector(arg.pass2RuntimeFilter)
```
Comment (fengttt) [inline pkg/sql/colexec/fuzzyfilter/types.go:117]: Can we simply get rid of putvector in proc?   Why proc need to know all these random stuff that has nothing to do with proc ...

Comment (jensenojs) [reply pkg/sql/colexec/fuzzyfilter/types.go:117]: @fengttt 这是一个session级别的vectorPool, 在算子里面要与vectorPool交互的话总要有个东西传递或者感知到它的, 不用proc的话, 这个指针要放哪呢

**Review by fengttt — APPROVED** [review]
why session level thing is in proc?    still a mistery.  but the function exists, so all the logic is lost.    anyway.

**Pull request merged**